plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/TestKafkaIntegrationPushDown.java

```diff
@@ -101,70 +100,60 @@ public void testPartitionPushDown()
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp = timestamp '%s'", topicNameCreateTime, recordMessage.getStartTime()), 0, session);
     }
 
     @Test
     public void testTimestampLogAppendModePushDown()
             throws Exception
     {
         RecordMessage recordMessage = createTimestampTestMessages(topicNameLogAppend);
-        DistributedQueryRunner queryRunner = getDistributedQueryRunner();
-        // ">= startTime" insure including index 2, "< endTime"  insure excluding index 4;
-        String sql = format(
-                "SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'",
-                topicNameLogAppend,
-                recordMessage.getStartTime(),
-                recordMessage.getEndTime());
+        Session session = getSession();
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp < timestamp '%s'", topicNameLogAppend, recordMessage.getEndTime()), 4, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp <= timestamp '%s'", topicNameLogAppend, recordMessage.getEndTime()), 4, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp > timestamp '%s'", topicNameLogAppend, recordMessage.getStartTime()), 997, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s'", topicNameLogAppend, recordMessage.getStartTime()), 998, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp between timestamp '%s' and timestamp '%s'", topicNameLogAppend, recordMessage.getStartTime(), recordMessage.getEndTime()), 2, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp >= timestamp '%s' and _timestamp < timestamp '%s'", topicNameLogAppend, recordMessage.getStartTime(), recordMessage.getEndTime()), 2, session);
+        assertProcessedInputPossitions(format("SELECT count(*) FROM default.%s WHERE _timestamp = timestamp '%s'", topicNameLogAppend, recordMessage.getStartTime()), 0, session);
+    }
 
+    private void assertProcessedInputPossitions(String sql, long expectedProcessedInputPositions, Session session)
+    {
+        DistributedQueryRunner queryRunner = getDistributedQueryRunner();
         assertEventually(() -> {
-            MaterializedResultWithQueryId queryResult = queryRunner.executeWithQueryId(getSession(), sql);
-            assertThat(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions())
-                    .isEqualTo(2);
+            MaterializedResultWithQueryId queryResult = queryRunner.executeWithQueryId(session, sql);
+            assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), expectedProcessedInputPositions);
```

Comment:
Let's add another `assertProcessedInputPositions` so that we can avoid passing the default session. 

```java
    private void assertProcessedInputPositions(String sql, long expectedProcessedInputPositions)
    {
        assertProcessedInputPositions(sql, expectedProcessedInputPositions, getSession());
    }

    private void assertProcessedInputPositions(String sql, long expectedProcessedInputPositions, Session session)
    {
        DistributedQueryRunner queryRunner = getDistributedQueryRunner();
        assertEventually(() -> {
            MaterializedResultWithQueryId queryResult = queryRunner.executeWithQueryId(session, sql);
            assertEquals(getQueryInfo(queryRunner, queryResult).getQueryStats().getProcessedInputPositions(), expectedProcessedInputPositions);
        });
    }
```

Reply:
OK, I added it.